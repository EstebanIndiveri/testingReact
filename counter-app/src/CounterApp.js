import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types'
const CounterApp = ({value}) => {

    const [count,setCounter]=useState(value);

    const handleAdd=()=>{
        setCounter(count +1 );
    } 
    const reset=()=>{
        setCounter(value);
    }
    const menos=()=>{
        setCounter(count -1 );

    }

    return ( 
        <Fragment>
            <h1>CounterApp</h1>
            <h1 style={{color:'violet'}}>{count}</h1>
            <button onClick={()=>handleAdd()} >+</button>
            <button onClick={()=>reset()} >Reset</button>
            <button onClick={()=>menos()} >-1</button>

        </Fragment>
     );
}

CounterApp.propTypes={
    value:PropTypes.number.isRequired
}
 
export default CounterApp;