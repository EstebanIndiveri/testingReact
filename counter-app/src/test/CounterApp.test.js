
import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

import CounterApp from '../CounterApp';

describe('pruebas <CounterApp/>', () => {
    test('<CounterApp correcto/>', () => {
        const wrapper=shallow(<CounterApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('100', () => {
        const wrapper=shallow(<CounterApp value={100}/>);
        const counterText=wrapper.find('.count').text().trim();
        expect(counterText).toBe('100');
    });
    
    test('incrementar boton +1', () => {
        const wrapper=shallow(<CounterApp/>);
        wrapper.find('button').at(0).simulate('click');
        const counterText=wrapper.find('.count').text().trim();
        expect(counterText).toBe('13');
    })
    
});
