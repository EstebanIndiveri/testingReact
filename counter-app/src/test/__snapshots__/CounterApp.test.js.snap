// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <CounterApp/> <CounterApp correcto/> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h1
    className="count"
    style={
      Object {
        "color": "violet",
      }
    }
  >
    12
  </h1>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
